### DISCUSSION TITLE
Which python libraries are included?

### DISCUSSION THREAD

**Discussion by olifante** [modular/modular#63] [Q&A] [answered]
The Mojo docs say the following:

> we’ve included a variety of popular Python packages, such as numpy, pandas, and matplotlib

However, I haven't found a way to list the other available packages. Some experimentation in the Playground shows that numba, scipy and sklearn are also included, but alas no PyTorch or Tensorflow:

```python
%%python
import pandas
print("pandas", pandas.__version__)
​
import numpy
print("numpy", numpy.__version__)
​
import matplotlib
print("matplotlib", matplotlib.__version__)
​
import sklearn
print("sklearn", sklearn.__version__)
​
import scipy
print("scipy", scipy.__version__)
​
import numba
print("numba", numba.__version__)

pandas 1.5.0
numpy 1.23.3
matplotlib 3.6.0
sklearn 1.1.2
scipy 1.9.1
numba 0.56.2
```

Is there a way to list the available libraries?

**Comment by TZDak** [comment]
I thought I had read something about support for something to do with PyTorch or Tensorflow being built in, but I wasn't sure exactly what... so I looked it up. Here's what I found at https://docs.modular.com/mojo/faq.html --- How does Mojo help with PyTorch and TensorFlow acceleration?
Mojo is a general purpose programming language, so it has no specific implementations for ML training or serving, although we use Mojo as part of the overall Modular AI stack. The [Modular Inference Engine](https://docs.modular.com/engine), for example, supports deployment of PyTorch and TensorFlow models, while Mojo is the language we use to write the engine’s in-house kernels.
While I know that doesn't answer your question, I hope it's helpful.

DAK

**Comment by wenijinew** [comment]
> Is there a way to list the available libraries?

As the document said, "you can import any other Python module.", so no limitation to use any Python modules, I guess.
https://docs.modular.com/mojo/programming-manual.html#python-integration

**Comment by timmermansjoy** [comment]
https://docs.modular.com/mojo/get-started.html#caveats indeed all python modules will be supported once the language is out. but for now in the notebook there is `numpy`, `pandas`and `matplotlib`

**Comment by toshvelaga** [comment, marked as answer]
Adding to this I think you can run this:


<img width="1075" alt="Screen Shot 2023-05-10 at 4 10 48 PM" src="https://github.com/modularml/mojo/assets/38474161/8207d17d-97be-4fb5-a6a5-4f7e040abe1a">

**Reply by AielloChan** [reply]
Can I manually install pkgs? such as PyTorch

**Reply by olifante** [reply]
Thanks a lot @toshvelaga, that clears it up. That's almost 200 packages, but many of them are dependencies of other packages. I was able to identify less than 50 top-level packages:

- altair
- backports.functools-lru-cache
- blinker
- bokeh
- Bottleneck
- brotlipy
- cached-property
- colorama
- Cython
- cytoolz
- dill
- distributed
- flit-core
- fqdn
- gmpy2
- greenlet
- h5py
- HeapDict
- imagecodecs
- importlib-resources
- ipympl
- isoduration
- jsonpointer
- jupyterhub
- jupyterlab
- lz4
- mamba
- munkres
- numba
- numexpr
- papermill
- pkgutil-resolve-name
- protobuf
- pycosat
- pycurl
- PyJWT
- PySocks
- scikit-image
- scikit-learn
- seaborn
- statsmodels
- sympy
- unicodedata2
- uri-template
- webcolors
- xlrd

**Reply by olifante** [reply]
@AielloChan unfortunately it's not currently possible to install packages in the playground. Quoting the docs:

> The Mojo environment does not have network access, so you cannot install other tools or Python packages. However, we’ve included a variety of popular Python packages, such as numpy, pandas, and matplotlib

https://docs.modular.com/mojo/get-started.html#caveats

**Comment by czheo** [comment]
```py
%%python
import os
os.system("pip list")
```

**Comment by iamtimdavis** [comment]
Please also suggest packages you would like to see added -> https://github.com/modularml/mojo/discussions/173
